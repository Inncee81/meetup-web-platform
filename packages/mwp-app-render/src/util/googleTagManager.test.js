import {
	getGoogleTagManagerSnippet,
	getDataLayerInitSnippet,
	gtmPush,
} from './googleTagManager';

describe('getGoogleTagManagerSnippet()', () => {
	it('matches snap', () => {
		expect(getGoogleTagManagerSnippet()).toMatchSnapshot();
	});
});

describe('getDataLayerInitSnippet()', () => {
	const MOCK_INIT_DATA = { foo: 'bar' };

	it('should match the snapshot without `data` passed', () => {
		expect(getDataLayerInitSnippet()).toMatchSnapshot();
	});

	it('should match the snapshot with some `data`', () => {
		expect(getDataLayerInitSnippet(MOCK_INIT_DATA)).toMatchSnapshot();
	});
});

describe('gtmPush()', () => {
	const MOCK_VARS = { foo: 'bar' };

	it('should push mocked variables to `dataLayer`', () => {
		global.window = {};
		window.dataLayer = [];

		gtmPush(MOCK_VARS);
		expect(window.dataLayer).toContain(MOCK_VARS);
	});
});
