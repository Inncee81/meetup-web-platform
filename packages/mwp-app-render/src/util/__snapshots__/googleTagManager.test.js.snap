// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDataLayerInitSnippet() should match the snapshot with some \`data\` 1`] = `"dataLayer=[{\\"foo\\":\\"bar\\"}]"`;

exports[`getDataLayerInitSnippet() should match the snapshot without \`data\` passed 1`] = `"dataLayer=[]"`;

exports[`getGoogleTagManagerSnippet() matches snap 1`] = `
"(function(w,d,s,l,i){
		w[l]=w[l]||[];
		w[l].push({'gtm.start':new Date().getTime(),event:'gtm.js'});
		var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
		j.async=true;j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl;
		f.parentNode.insertBefore(j,f);
	})(window,document,'script','dataLayer','GTM-W9W847');"
`;
